File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.go
func removeDuplicates(nums []int) int {
    i, j, res := 0, 1, 1
    
    for j<len(nums) {
        if nums[i] != nums[j] {
            res++
            i += 1
            nums[i] = nums[j]
        }
        j++
    }
    return res
}

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int i=0, j=0, res=1;
        while(j<nums.size()) {
            if(nums[i] != nums[j]) {
                nums[++i] = nums[j];
                res++;
            }    
            j++;
        }
        return res;
    }
};